cdap-notifications/src/main/java/co/cask/cdap/notifications/service/BasicNotificationContext.java

```diff
@@ -51,7 +51,7 @@ public boolean execute(TxRunnable runnable, TxRetryPolicy policy) {
         final TransactionContext context = new TransactionContext(transactionSystemClient);
         try {
           context.start();
-          runnable.run(new DynamicDatasetContext(namespaceId, context, dsFramework,
+          runnable.run(new DynamicDatasetContext(namespaceId, context, null, dsFramework,
```

Comment:
This is a system service that currently has no metric context. I think it goes beyond the scope of this PR to change that. Opened https://issues.cask.co/browse/CDAP-3114